src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp

```diff
@@ -0,0 +1,232 @@
+}
+
+
+bool ThreeAxisForceTorqueContactSensor::setParentLink(const std::string& parent)
+{
+    this->pimpl->parent_link_name = parent;
+    return true;
+}
+
+bool ThreeAxisForceTorqueContactSensor::setParentLinkIndex(const LinkIndex &parent_index)
+{
+    this->pimpl->parent_link_index = parent_index;
+    return true;
+
+}
+
+std::string ThreeAxisForceTorqueContactSensor::getParentLink() const
+{
+    return(this->pimpl->parent_link_name);
+}
+
+int ThreeAxisForceTorqueContactSensor::getParentLinkIndex() const
+{
+    return(this->pimpl->parent_link_index);
+}
+
+bool ThreeAxisForceTorqueContactSensor::isValid() const
+{
+    if( this->getName() == "" )
+    {
+        return false;
+    }
+
+    if( this->pimpl->parent_link_index < 0 )
+    {
+        // Return false because the links is not appropriately setted
+        return false;
+    }
+
+    return true;
+}
+
+Sensor* ThreeAxisForceTorqueContactSensor::clone() const
+{
+    return (Sensor *)new ThreeAxisForceTorqueContactSensor(*this);
+}
+
+bool ThreeAxisForceTorqueContactSensor::updateIndices(const Model& model)
+{
+    iDynTree::LinkIndex linkNewIndex = model.getLinkIndex(this->pimpl->parent_link_name);
+
+    if( linkNewIndex == iDynTree::LINK_INVALID_INDEX )
+    {
+        return false;
+    }
+
+    this->pimpl->parent_link_index = linkNewIndex;
+
+    return true;
+}
+
+std::string ThreeAxisForceTorqueContactSensor::getName() const
+{
+    return this->pimpl->name;
+}
+
+SensorType ThreeAxisForceTorqueContactSensor::getSensorType() const
+{
+    return THREE_AXIS_FORCE_TORQUE_CONTACT;
+}
+
+Transform ThreeAxisForceTorqueContactSensor::getLinkSensorTransform() const
+{
+    return(this->pimpl->link_H_sensor);
+}
+
+void ThreeAxisForceTorqueContactSensor::setLoadCellLocations(std::vector<Position> &loadCellLocations)
+{
+    pimpl->m_loadCellLocations = loadCellLocations;
+}
+
+std::vector<Position>  ThreeAxisForceTorqueContactSensor::getLoadCellLocations()
+{
+    return pimpl->m_loadCellLocations;
+}
+
+Vector3  ThreeAxisForceTorqueContactSensor::computeThreeAxisForceTorqueFromLoadCellMeasurements(VectorDynSize& loadCellMeasurements)
+{
+    if (loadCellMeasurements.size() != pimpl->m_loadCellLocations.size())
+    {
+        reportError("ThreeAxisForceTorqueContactSensor","computeThreeAxisForceTorqueFromLoadCellMeasurements", "loadCellMeasurements has the wrong size");
+        Vector3 zero; zero.zero();
+        return zero;
+    }
+
+    Wrench totalWrench;
+    totalWrench.zero();
+
+    for (size_t i=0; i < loadCellMeasurements.size(); i++)
+    {
+        // We assume that the total forcetorque is given by the sum of pure forces on the z axis measured by the load cells
+        Wrench loadCellWrench = Wrench(LinearForceVector3(0.0, 0.0, loadCellMeasurements(i)*1.0), AngularForceVector3(0.0, 0.0, 0.0));
+        Transform sensor_H_loadCell(Rotation::Identity(), pimpl->m_loadCellLocations[i]);
+        totalWrench = totalWrench + sensor_H_loadCell*loadCellWrench;
```

Comment:
nitpick: I usually find more compact (but also readable) the `+=` operator

Reply:
There is no `Wrench::operator+=` at the moment. : )

Reply:
I did't realise it was on the Wrench object. Ok :)